### ISSUE TITLE
Create kube config for ALL clusters user has access to

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#13853]
kube config suppports having more than one cluster defined and switching "contexts" we should give the user the ability to generate a kubeconfig that has all the clusters they have access to in it so that they can quickly and easily switch between clusters

**Comment by pcornelissen** [comment]
Also it would be neat to be able to create the kube config for another user when you are the clusteradmin. Because then I can create the deployment users with the appropriate permissions and then create the config for my deployment pipeline with ease

**Comment by nlowe** [comment]
This would be especially helpful now that we have to manually tweak the usernames in the kubeconfig if we are switching between clusters in 2.2+ (#19342)

**Comment by bgagnon** [comment]
This would be useful both from the Rancher UI and the Rancher CLI.

For the CLI, it's somewhat achievable today by taking the outputs of`rancher clusters kubeconfig` on every cluster, then using something like https://github.com/corneliusweig/konfig to merge them.

**Comment by asemen2608** [comment]
Could this be used to download in UI the kubeconfig per cluster?  Like needing in #28503

**Comment by asemen2608** [comment]
In which Rancher release can we expect to see this feature implemented?

**Comment by lukastopiarz** [comment]
Hello,
What's the progress, please? We would love a "multi-kubeconfig" file feature.

**Comment by n89cz** [comment]
This feature would be really great to have.

**Comment by bjd2385** [comment]
I agree, in the meantime I'm going to have to write a work-around script that merges dumped configs instead.

**Comment by bjd2385** [comment]
For anyone else who would find it useful, I've written a script that can download and merge all your clusters' kube configs on your local machine from a Rancher instance.

https://gist.github.com/bjd2385/b9425e3cad33f871e278d135ea8bad1a

Developers could place an `echo all | ./rancher.ssh` in their `~/.profile` so they can context switch in their IDE of choice (like Lens) after login.

```shell
Select a cluster:

(1)  production
(2)  load
(3)  rancher
(4)  development
(5)  staging
(6)  local
(7)  all

Select a cluster (by name or index): 
WARNING: Defaulting to development
Info: Attempting to create ~/.kube/config for cluster "development"
Info: Successfully switched context to cluster "development"
```

**Comment by murand78** [comment]
Solved using ranche-cli and [konfig](https://github.com/corneliusweig/konfig)

/tmp % rancher clusters ls --format 'rancher cluster kubeconfig {{.Cluster.ID}} > {{.Cluster.Name}}.kubeconfig'
(execute output) 

/tmp % rancher clusters ls --format 'konfig import --save {{.Cluster.Name}}.kubeconfig' 
(execute output) 

then switch context using https://github.com/ahmetb/kubectx
